packages/flutter_tools/lib/src/cmake.dart

```diff
@@ -25,19 +30,43 @@ String _escapeBackslashes(String s) {
   return s.replaceAll(r'\', r'\\');
 }
 
+Version _parseBuildName(String buildName) {
+  try {
+    return Version.parse(buildName);
+  } on FormatException {
+    globals.printWarning('Warning: could not parse build name $buildName, defaulting to 1.0.0.');
+
+    return Version(1, 0, 0);
+  }
+}
+
 /// Writes a generated CMake configuration file for [project], including
 /// variables expected by the build template and an environment variable list
 /// for calling back into Flutter.
-void writeGeneratedCmakeConfig(String flutterRoot, CmakeBasedProject project, Map<String, String> environment) {
+void writeGeneratedCmakeConfig(
+  String flutterRoot,
+  CmakeBasedProject project,
+  BuildInfo buildInfo,
+  Map<String, String> environment) {
   // Only a limited set of variables are needed by the CMake files themselves,
   // the rest are put into a list to pass to the re-entrant build step.
   final String escapedFlutterRoot = _escapeBackslashes(flutterRoot);
   final String escapedProjectDir = _escapeBackslashes(project.parent.directory.path);
+  final FlutterManifest manifest = project.parent.manifest;
+  final String buildName = buildInfo.buildName ?? manifest.buildName ?? '1.0.0';
+  final String buildNumber = buildInfo.buildNumber ?? manifest.buildNumber ?? '1';
```

Comment:
Won't this make `0.x.y` and `0.x.y+1` the same Windows version?

Why are we unconditionally adding a build number? Is that required on Windows?

Reply:
> Won't this make `0.x.y` and `0.x.y+1` the same Windows version?

In this scenario, those versions would produce the same *product* version but different *file* versions. 

> Why are we unconditionally adding a build number? Is that required on Windows?

It is required in scenarios where you want an installer to update files without changing the product version.

Reply:
Do you have any docs you could point me to here? I'm not understanding why it's desirable to treat two different Dart versions as having the same product versions.

Reply:
Ah it turns out I goofed: the build "number" isn't actually a number, it can contain non-numeric character. My current PR doesn't handle this properly. I will update this PR to ignore the build number entirely for `flutter windows build`, and instead, use only the build name. This will make the product and file versions the same value..

> Do you have any docs you could point me to here? I'm not understanding why it's desirable to treat two different Dart versions as having the same product versions.

I'm having a hard time finding relevant docs, but this is a pattern I've seen from my .NET days where folks want to patch `.dll`s that are binary compatible. For example, there's a [Global Assembly Cache (GAC)](https://docs.microsoft.com/en-us/dotnet/framework/app-domains/gac) (used in legacy .NET scenarios) where updating an assembly requires re-using a previous version number. In these scenarios, folks have two versions to separate the "branding" version from the actual version.

Reply:
I updated the PR to ensure different Dart versions result in different Windows version strings, and, to no longer unconditionally add a build number. Please take another look and let me know if you have additional feedback!

Reply:
At a minimum we should be emitting a prominent warning in the case where we can't make translate some portion cleanly to a Windows version number.

Reply:
We discussed this on Discord a bit earlier today and the suggestion was that we emit the warning, but if we're finding people filing issues about it, an option might be to add the option for a platform-specific version in the pubspec that users could set as an override, which would also silence the warning.

Reply:
Thanks for the suggestions everyone! If the build number is numeric, it is now translated to a Windows version number. Otherwise, a warning is outputted on Windows. If this warning proves to be cumbersome we will add a platform-specific version override as suggested by Chris above. I'll go ahead and mark this as resolved, but please let me know if you have additional feedback!